substrate-archive/src/archive.rs

```diff
@@ -73,116 +128,192 @@ where
-	/// and directory where the WASM runtimes are stored.
-	pub wasm_tracing: Option<TracingConfig>,
+	config: ArchiveConfig,
 }
 
 impl<B, R, D, DB> Default for ArchiveBuilder<B, R, D, DB> {
 	fn default() -> Self {
-		let num_cpus = num_cpus::get();
-		Self {
-			_marker: PhantomData,
-			chain_spec: None,
-			chain_data_path: None,
-			pg_url: None,
-			cache_size: 128,                  // 128 MB
-			block_workers: num_cpus,          // the number of logical cpus in the system
-			wasm_pages: 64 * num_cpus as u64, // 64 * (number of logic cpu's)
-			max_block_load: 100_000,          // 100_000 blocks to index at once
-			wasm_tracing: None,
-		}
+		Self { _marker: PhantomData, config: ArchiveConfig::default() }
 	}
 }
 
 impl<B, R, D, DB> ArchiveBuilder<B, R, D, DB> {
-	/// Specify a chain spec for storing metadata about the running archiver
-	/// in a persistant directory.
+	/// Creates a archive builder with the given config.
+	pub fn with_config(config: Option<ArchiveConfig>) -> Self {
+		if let Some(config) = config {
+			Self { _marker: PhantomData, config }
+		} else {
+			Self::default()
+		}
+	}
+
+	/// Specify a chain spec name and id for storing metadata about the running archiver
+	/// in a persistent directory.
 	///
 	/// # Default
-	/// Defaults to storing metadata in a temporary directory.
+	/// defaults to storing metadata in a temporary directory.
 	pub fn chain_spec(mut self, spec: Box<dyn ChainSpec>) -> Self {
-		self.chain_spec = Some(spec);
+		self.config.chain.spec = Some(ChainSpecConfig { name: spec.name().into(), id: spec.id().into() });
 		self
 	}
 
 	/// Set the chain data backend path to use for this instance.
 	///
 	/// # Default
 	/// defaults to the environment variable CHAIN_DATA_DB
-	pub fn chain_data_path<S: Into<PathBuf>>(mut self, path: Option<S>) -> Self {
-		self.chain_data_path = path.map(Into::into);
+	pub fn chain_data_path<S: Into<PathBuf>>(mut self, path: S) -> Self {
+		self.config.chain.data_path = Some(path.into());
+		self
+	}
+
+	/// Set the amount of cache RocksDB should keep.
+	///
+	/// # Default
+	/// defaults to 128MB
+	pub fn cache_size(mut self, cache_size: usize) -> Self {
+		self.config.chain.cache_size = cache_size;
+		self
+	}
+
+	/// Set the path of RocksDB secondary directory according to the chain spec.
+	///
+	/// # Default
+	/// defaults to storing metadata in a temporary directory.
+	pub fn rocksdb_secondary_path<S: Into<PathBuf>>(mut self, path: S) -> Self {
+		self.config.chain.rocksdb_secondary_path = Some(path.into());
 		self
 	}
 
 	/// Set the url to the Postgres Database
 	///
 	/// # Default
 	/// defaults to value of the environment variable DATABASE_URL
-	pub fn pg_url<S: Into<String>>(mut self, url: Option<S>) -> Self {
-		self.pg_url = url.map(Into::into);
+	pub fn pg_url<S: Into<String>>(mut self, url: S) -> Self {
+		self.config.database = Some(DatabaseConfig { url: url.into() });
 		self
 	}
 
-	/// Set the amount of cache Rocksdb should keep.
```

Comment:
Maybe I missed it, but where do we set the cache size now?

Reply:
the `cache_size` method of `ArchiveBuilder`